org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/adapters/WorkDoneProgressNotificationAdapter.java

```diff
@@ -0,0 +1,175 @@
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import com.google.gson.stream.MalformedJsonException;
+
+/**
+ * A type adapter for class which implements WorkDoneProgressNotification interface.
+ */
+public class WorkDoneProgressNotificationAdapter extends TypeAdapter<WorkDoneProgressNotification> {
+
+	public static class Factory implements TypeAdapterFactory {
+
+		@SuppressWarnings("unchecked")
+		@Override
+		public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> typeToken) {
+			Class<?> rawType = typeToken.getRawType();
+			if (!WorkDoneProgressNotification.class.isAssignableFrom(rawType)) {
+				return null;
+			}
+			return (TypeAdapter<T>) new WorkDoneProgressNotificationAdapter();
+		}
+
+	}
+
+	public WorkDoneProgressNotificationAdapter() {
+	}
+
+	@Override
+	public WorkDoneProgressNotification read(JsonReader in) throws IOException {
+		if (in.peek() == JsonToken.NULL) {
+			in.nextNull();
+			return null;
+		}
+		
+		in.beginObject();
+		String kind = null, message = null, title = null;
+		Boolean cancellable = null;
+		Integer percentage = null;
+		try {
+			
+			while (in.hasNext()) {
+				String name = in.nextName();
+				switch (name) {
+				case "kind": {
+					kind = in.nextString();
+					break;
+				}
+				case "title": {
+					title = in.nextString();
+					break;
+				}
+				case "message": {
+					message = in.nextString();
+					break;
+				}
+				case "cancellable": {
+					cancellable = in.nextBoolean();
+					break;
+				}
+				case "percentage": {
+					percentage = in.nextInt();
+					break;
+				}
+				default:
+					in.skipValue();
+				}
+			}
+			in.endObject();
+			return createNotification(kind, message, title, cancellable, percentage);
+			
+		} catch (JsonSyntaxException | MalformedJsonException | EOFException exception) {
+			if (kind != null) {
+				throw new JsonParseException(exception);
+			} else {
+				throw exception;
+			}
+		}
+	}
+
+	private WorkDoneProgressNotification createNotification(String kind, String message, String title,
+			Boolean cancellable, Integer percentage) throws MalformedJsonException {
+		if(kind == null) {
+			throw new MalformedJsonException("Kind of progress notification is empty");
+		}
+		switch (kind) {
+		case "begin":
+			WorkDoneProgressBegin begin = new WorkDoneProgressBegin();
+			begin.setMessage(message);
+			begin.setCancellable(cancellable);
+			begin.setPercentage(percentage);
+			begin.setTitle(title);
+			return begin;
+		case "report":
+			WorkDoneProgressReport report = new WorkDoneProgressReport();
+			report.setMessage(message);
+			report.setCancellable(cancellable);
+			report.setPercentage(percentage);
+			return report;
+		case "end":
+			WorkDoneProgressEnd end = new WorkDoneProgressEnd();
+			end.setMessage(message);
+			return end;
+		default:
+			throw new MalformedJsonException("Kind of progress notification is unknown :"+kind);
```

Comment:
The colon at the end of the error message should be before the space and not after the space.